src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1107,6 +1107,7 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                     HashSet<LazyStringValue> docCountersToRecreate = null;
                     var handledAttachmentStreams = new HashSet<Slice>(SliceComparer.Instance);
                     context.LastDatabaseChangeVector ??= DocumentsStorage.GetDatabaseChangeVector(context);
+                    database.DocumentsStorage.AttachmentsStorage.AttachmentStreamsToDeleteAtTheEndOfBatch.Clear();
```

Comment:
no, this can't be in the shard instance of the `AttachmentsStorage`, it has to be local.

Reply:
maybe on the `_replicationInfo`

Reply:
Moved